src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -0,0 +1,433 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Bipartite.AdjacencyMap
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- __Alga__ is a library for algebraic construction and manipulation of graphs
+-- in Haskell. See <https://github.com/snowleopard/alga-paper this paper> for
+-- the motivation behind the library, the underlying theory, and
+-- implementation details.
+--
+-- This module defines the 'AdjacencyMap' for bipartite graphs data type and
```

Comment:
I think you meant `This module defines the 'AdjacencyMap' data type for bipartite graphs and`